osu.Game/Updater/SimpleUpdateManager.cs

```diff
@@ -74,37 +73,8 @@ protected override async Task<bool> PerformUpdateCheck()
             return false;
         }
 
-        private string getBestUrl(GitHubRelease release)
-        {
-            GitHubAsset bestAsset = null;
-
-            switch (RuntimeInfo.OS)
-            {
-                case RuntimeInfo.Platform.Windows:
-                    bestAsset = release.Assets?.Find(f => f.Name.EndsWith(".exe", StringComparison.Ordinal));
-                    break;
-
-                case RuntimeInfo.Platform.macOS:
-                    string arch = RuntimeInformation.OSArchitecture == Architecture.Arm64 ? "Apple.Silicon" : "Intel";
-                    bestAsset = release.Assets?.Find(f => f.Name.EndsWith($".app.{arch}.zip", StringComparison.Ordinal));
-                    break;
-
-                case RuntimeInfo.Platform.Linux:
-                    bestAsset = release.Assets?.Find(f => f.Name.EndsWith(".AppImage", StringComparison.Ordinal));
-                    break;
+        protected virtual void UpdateActionOnClick(GitHubRelease release) => host.OpenUrlExternally(GetBestURL(release));
 
-                case RuntimeInfo.Platform.iOS:
-                    // iOS releases are available via testflight. this link seems to work well enough for now.
-                    // see https://stackoverflow.com/a/32960501
-                    return "itms-beta://beta.itunes.apple.com/v1/app/1447765923";
-
-                case RuntimeInfo.Platform.Android:
-                    // on our testing device this causes the download to magically disappear.
-                    //bestAsset = release.Assets?.Find(f => f.Name.EndsWith(".apk"));
-                    break;
-            }
-
-            return bestAsset?.BrowserDownloadUrl ?? release.HtmlUrl;
-        }
+        protected virtual string GetBestURL(GitHubRelease release) => release.HtmlUrl;
```

Comment:
Why virtual? Why is both this and `UpdateActionOnClick()` virtual? Means that overriding the latter potentially makes this completely useless.